### ISSUE TITLE
Postback vs. waiting

### ISSUE THREAD

**Issue by yoavweiss** [ResponsiveImagesCG/Sizer-Soze-frontend#11]
I've set up the service part, and significantly reduced the delay time it takes to get a response (currently it's around 30-60 seconds, depends on site).

Which got me thinking. If we decide that the front-end would simply wait until the service responds, we can have a static, JS-in-the-browser-only front-end, which will significantly ease its deployment.
Alternatively, we can do postback responses, which require a more complex front-end, but I'm not sure what are the benefits of that.

@jlembeck - what do you say?

**Comment by yoavweiss** [comment]
Currently I got the service to do both. It postbacks the JSON to the given URL, and returns it as the response (after blocking during the execution time).

**Comment by jefflembeck** [comment]
@yoavweiss So, the reason I initially wanted them separate like this and to have a postback was because:
1. As @wilto mentioned, huge fan of PE. Having this work without JS is something that's totally achievable and I dig it. If it's not in the cards for the project, I get that for certain.
2. I was honestly a bit worried about how much of a beating your service would get. If we have a full stack over here, I can cache responses for a set period of time, so in case somebody in some office sends out a link telling everybody to check their site, you wouldn't tip over.

Those are the positives.

Negatives, of course, are -- since we're not using some sort of auto-setup with chef/vagrant, you'd have to set up your environment to support a node app behind nginx and a db of some sort (key/val works for me in this case, it's mostly for caching, Redis tends to do this part well).

I'll let you make the call on this, as I might be prematurely optimizing quite a bit over here.

**Comment by yoavweiss** [comment]
@jlembeck I got the service to do both, so let's start with a JS based, static app, that wait on the request and gets the JSON including all the viewports as a result, then move to something with a smarter backend that can work JS free.
WDYT?

**Comment by jefflembeck** [comment]
@yoavweiss Single page is easy enough and something we can get off of the ground pretty quickly. Moving backwards isn't super simple, but something we can definitely do.

@garann Any suggestions if we're going to do this pure front-end? It's pretty simple, functionality-wise, so I don't think we're going to need anything like Backbone, but if we will, I'm all ears.

@Wilto @hellogeri So we have our original design in place right now, do we have something for after the data is received back/ where that data should be display/should look like?

**Comment by garann** [comment]
@jlembeck Not entirely sure what you're asking, but to send a request and return a response, no, I can't think of any reason you'd need a framework. Seem to be missing the place in the code where the work is actually getting done, however, so please tell me if there's more to it than that.

ETA: oh n/m, it's in that other issue. :) Yeah, unless that response data will continue to be worked with and persisted somewhere once it comes in, frameworkless sounds good to me.

**Comment by Wilto** [comment]
@jlembeck: Nothing in the way of a response page yet. If you wanna just blurt it all out into an unstyled table, we can wing the design around it.

**Comment by hellogeri** [comment]
@starburst1977 Hey—did you ever work up a design for a result page? I know the project sort of went cold for a bit, but we're going to need one, as mentioned above :)

**Comment by starburst1977** [comment]
@jlembeck @hellogeri No, there is no results page so far. Just go ahead like @Wilto said and dump it inside some simple HTML. I will take care of it as soon as I can see what it will look like.

**Comment by jefflembeck** [comment]
@starburst1977 @hellogeri @Wilto I can definitely get the data into some type of table.

As far as interaction goes, we have inputs right now, but nothing to send the action (I suppose an 'enter' on the url could work... so I'll start with that). Button preferences?

**Comment by jefflembeck** [comment]
We decided to go with waiting, the table is now created in the fourth-content/sizer-results area. Will make another ticket for styling that.
